OpenEdX/Managers/PushNotificationsManager/PushNotificationsManager.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  PushNotificationManager.swift
+//  OpenEdX
+//
+//  Created by Anton Yarmolenka on 24/01/2024.
+//
+
+import Foundation
+import Core
+import UIKit
+import Swinject
+
+public protocol PushNotificationsProvider {
+    func registerWithDeviceToken(deviceToken: Data)
+    func didFailToRegisterForRemoteNotificationsWithError(error: Error)
```

Comment:
It would be nice if both methods followed the same naming convention either with did or without did.